lib/Converter/FunctionConverter.cpp

```diff
@@ -59,6 +62,99 @@ Node &FunctionConverter::morphNode(Node &node) { return node; }
 
 void FunctionConverter::postProcessing(Node &node) {}
 
+void FunctionConverter::convertOutputs(Node &node) {
+  using FunctionAndValIdx = std::pair<Function *, unsigned>;
+  llvm::DenseMap<FunctionAndValIdx, NodeValue> functionAndValToConversion;
+  for (unsigned idx = 0, end = node.getNumResults(); idx != end; ++idx) {
+    NodeValue val = node.getNthResult(idx);
+    TypeRef targetTy = getTargetTypeForOutput(val);
+    if (!targetTy || targetTy == val.getType()) {
+      continue;
+    }
+    // convert the node and create a conversion to keep the users happy.
+    assert(targetTy->dims() == val.getType()->dims() &&
+           "Conversion does not preserve shape");
+    TypeRef origTy = val.getType();
+    // Fake the morphing of the node so that the creation
+    // of the conversion works properly.
+    val.setType(targetTy);
+    // Store the users in a temporary object because setOperand
+    // will invalidate the iterator.
+    llvm::SmallVector<NodeUse, 4> users(val.getUsers().begin(),
+                                        val.getUsers().end());
+    // We cannot use replaceAllUsesWith here because:
```

Comment:
I noticed this "4 users" before but forget to ask -- Just wondering how this "4" comes from :) Thanks!